### DISCUSSION TITLE
How to preserve SMTP server configuration?

### DISCUSSION THREAD

**Discussion by villegas97** [PHPMailer/PHPMailer#2807] [Q&A]
I need to send a few hundreds of emails (I have only around 10 addresses, but each of them can have a few dozens of emails to be sent individually) every day, but in my code I include the server configuration for every email I send, which is giving me problems with Google (which is my provider), because they limit the number of connections made to the server. 

I have a while loop, which consumes my phpmailer class, which at the same time, sends the emails. 
I made a script test to see if I could leave the server configuration out of the loop and send 3 emails without having to use the entire code for every email, but it didn't work, only sent 1. This is my test script:

```
require '../lib/PHPMailer52/PHPMailerAutoload.php';

$mail = new PHPMailer(true);

$mail->SMTPDebug = 0;                                     
$mail->isSMTP();                                          
$mail->Host = 'smtp.gmail.com';                       
$mail->SMTPAuth = true;                                       
$mail->Username = 'xxxxxx';             
$mail->Password = 'xxxxx';                               
$mail->SMTPSecure = "ssl";                                    
$mail->Port = 465;                                            
$mail ->CharSet = "UTF-8";
$mail->setFrom('xxxxxx', 'xxx');

for($i=0; $i<3; $i++){
    $asunto_user="Assigned Activity: ".$i."";
    $reporte_user="Empty";
    try {
        $mail->addAddress("xxxxxx", "xx");                      
        $mail->addReplyTo("xxxxxx", 'xxx');
        $mail->isHTML(true);                                     
        $mail->Subject = $asunto_user;
        $mail->Body    = $reporte_user;
        
        $mail->send();
        return "ok";
      }//try
    catch (Exception $e){
          echo 'No se ha podido enviar el Mensaje. Error: ', $mail->ErrorInfo;
          return "error";
          
    }
}
```
What could I do to not connect to the smtp server before every email? It would be better if I only connect once for every address.

**Comment by Synchro** [comment]
I recommend reading [the docs for sending to lists](https://github.com/PHPMailer/PHPMailer/wiki/Sending-to-lists), and base your code on [the mailing list example](https://github.com/PHPMailer/PHPMailer/blob/master/examples/mailing_list.phps), which does exactly what you ask, making use of keep-alive.
